### DISCUSSION TITLE
Create a plugin that adds new config values!

### DISCUSSION THREAD

**Discussion by smithliam** [tailwindlabs/tailwindcss#9263] [Help] [answered]
Hi

This link wan't helpful:
https://tailwindcss.com/docs/plugins

Let's say this is my `tailwind.config.cjs` file:

```js
module.exports = {
  theme: {
    extend: {
      width: {
        '25%': '25%',
        '50%': '50%',
      },
    },
  },
}
```

How can I create a plugin that does the same? I want to add new/more default values and I also want the user to be able to edit the values or add new values. I want to add those values to 1 or a few properties like (width, minWidth, maxWidth, height, minHeight, maxHeight and etc), and I also want the user to be able to customize the properties!

**Comment by wongjn** [comment]
So it seems like you want to create a plugin that provides [dynamic utilities](https://tailwindcss.com/docs/plugins#dynamic-utilities). How would you envisage the user to configure and use your plugin? Are you looking to supersede the built-in width, minWidth, maxWidth, height, minHeight, maxHeight, etc plugins with your own? Could you share some pseudo-code as examples of usage and output?

**Comment by crswll** [comment, marked as answer]
It sounds like maybe you want presets: https://tailwindcss.com/docs/presets
